### PR TITLE
fix isManagerAsync(), remove recursion, add memoryCacheHelper

### PR DESCRIPTION
**Pull request by ndoriomedov**
https://trello.com/c/6qJEHECu

### CODE REVIEW

**Review by vorobevamv — APPROVED** [review]

File: src/DepartmentService.Broker/Helpers/Branch/DepartmentBranchHelper.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+      while (childrenIdsStack.Any())
+      {
+        Guid parentId = childrenIdsStack.Pop();
+
+        if (childrenDictionary.TryGetValue(parentId, out IEnumerable<Guid> currentChildrenIds))
+        {
+          childrenIds.AddRange(currentChildrenIds);
+
+          foreach (var childId in currentChildrenIds)
```
Comment (VLiamin) [inline src/DepartmentService.Broker/Helpers/Branch/DepartmentBranchHelper.cs:49]: May be remove var?

File: src/DepartmentService.Broker/Helpers/Branch/DepartmentBranchHelper.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    public async Task<DepartmentUserRole?> GetDepartmentUserRole(Guid userId, Guid departmentId)
+    {
+      DbDepartmentUser dbUser = await _departmentUserRepository.GetAsync(userId);
+
+      if (dbUser is null || dbUser.DepartmentId == departmentId)
+      {
+        return (DepartmentUserRole?)dbUser?.Role;
+      }
+
+      return GetChildrenIds(await _memoryCacheHelper.GetDepartmentsTreeAsync(), dbUser.DepartmentId)?.Contains(departmentId) == true
```
Comment (VLiamin) [inline src/DepartmentService.Broker/Helpers/Branch/DepartmentBranchHelper.cs:68]: May be remove == true?

File: src/DepartmentService.Mappers/Models/DepartmentsTreeInfoMapper.cs [hunk 3]
```diff
@@ -10,14 +10,33 @@ public class DepartmentsTreeInfoMapper : IDepartmentsTreeInfoMapper
...
+
+      while (childrenStack.Any())
+      {
+        DepartmentsTreeInfo currentParent = childrenStack.Pop();
+
+        if (childrenDictionary.TryGetValue(currentParent.Id, out var currentChildren))
+        {
+          currentParent.Children = currentChildren;
+
+          foreach (var child in currentChildren)
```
Comment (VLiamin) [inline src/DepartmentService.Mappers/Models/DepartmentsTreeInfoMapper.cs:32]: And here

**Review by VLiamin — APPROVED** [review]

File: src/DepartmentService.Broker/Helpers/Branch/DepartmentBranchHelper.cs (hunk 2 above)
Comment (ndoriomedov) [reply src/DepartmentService.Broker/Helpers/Branch/DepartmentBranchHelper.cs:68]: can't, because GetChildrenIds() can return null

**Pull request merged**